### PR TITLE
[#328] Default Safe variable is incorrect in many cases

### LINKED ISSUE #328: DefaultSafe variable is incorrect in many cases
# 🐛 Summary #
The `DefaultSafe` is a boolean variable in the Rego, intended to indicate whether the default value in the GWS admin center complies with the baseline. However, there are many cases where the `DefaultSafe` listed in the Rego is incorrect.

Cases where `DefaultSafe` should be set to true:
- GWS.GROUPS.1.1, GWS.GROUPS.2.1, GWS.GROUPS.3.1, GWS.GROUPS.6.1
- GWS.MEET.4.1, GWS.MEET.5.1
- GWS.DRIVEDOCS.1.3, GWS.DRIVEDOCS.1.8, GWS.DRIVEDOCS.2.3, GWS.DRIVEDOCS.2.4, GWS.DRIVEDOCS.3.1
- GWS.CLASSROOM.1.1, GWS.CLASSROOM.1.2
- GWS.CHAT.1.1, GWS.CHAT.3.1
- GWS.CALENDAR.1.1, GWS.CALENDAR.2.1, GWS.CALENDAR.3.1, GWS.CALENDAR.4.1
- GWS.GMAIL.5.1, GWS.GMAIL.5.2, GWS.GMAIL.5.4, GWS.GMAIL.6.1, GWS.GMAIL.6.2, GWS.GMAIL.6.3, GWS.GMAIL.6.4, GWS.GMAIL.7.1, GWS.GMAIL.7.2, GWS.GMAIL.7.3, GWS.GMAIL.7.7, GWS.GMAIL.8.1, GWS.GMAIL.12.1, GWS.GMAIL.13.1, GWS.GMAIL.14.1, GWS.GMAIL.15.1
- GWS.COMMONCONTROLS.5.1, GWS.COMMONCONTROLS.5.4, GWS.COMMONCONTROLS.5.5, GWS.COMMONCONTROLS.10.3, GWS.COMMONCONTROLS.11.2, GWS.COMMONCONTROLS.15.2

Cases where `DefaultSafe` should be set to false:
- GWS.CLASSROOM.4.1
- GWS.COMMONCONTROLS.12.1

## To reproduce ##
While inside a brand new, default state GWS environment, check the default values.

## Expected behavior ##
The `DefaultState` coded in the Rego to be correct in all cases.

### PR DESCRIPTION
**Pull request by snarve**
## 🗣 Description ##

Default Safe variable is incorrect in many cases

Closes #328 

### 💭 Motivation and context

The Default Safe variable was assigned an incorrect value in many of the baselines and this was resulting in incorrect conditional checks when there are no log events present for a given policy.

## 🧪 Testing

- Testing may involve testing in a default setting environment (potentially new environment) where the settings/controls are unchanged and there are no log events.

- Another work around to test this is to delete the logs from the 'ProviderSettingExport.json' file, that way no log events will be present and the DefaultSafe value can be tested
- This update touched many of the baseline policies and corresponding unit tests, so ensure everything is working as expected

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] If applicable, *All* future TODOs are captured in issues, which are referenced in the PR description.
- [x] The relevant issues PR resolves are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels have been added.
- [x] I have read and agree to the [CONTRIBUTING.md](https://github.com/cisagov/ScubaGoggles/blob/main/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge Checklist

- [x] This PR has been smoke tested to ensure main is in a functional state when this PR is merged.
- [ ] Squash all commits into one PR level commit using the `Squash and merge` button.

## ✅ Post-merge Checklist

- [x] Delete the branch to clean up.
- [x] Close issues resolved by this PR if the [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) did not activate.

### CODE REVIEW

**Comment by snarve** [comment]
@adhilto @mitchelbaker-cisa @LaurenBassett this PR is ready for review.

File: rego/Groups.rego [hunk 1]
```diff
@@ -44,8 +44,8 @@ tests contains {
     "NoSuchEvent": true
 }
 if {
-    DefaultSafe := false
-    Events := utils.FilterEventsOU(LogEvents, "GroupsSharingSettingsProto collaboration_policy", utils.TopLevelOU)
+    DefaultSafe := true
+    Events := utils.FilterEvents(LogEvents, "GroupsSharingSettingsProto collaboration_policy", utils.TopLevelOU)
```
Comment (adhilto) [inline rego/Groups.rego:48]: Why did this line change from `FilterEventsOU()` to `FilterEvents()`? The `FilterEvents()` function is (unofficially) deprecated.

File: rego/Groups.rego [hunk 2]
```diff
@@ -357,8 +357,8 @@ tests contains {
     "NoSuchEvent": true
 }
 if {
-    DefaultSafe := false
-    Events := utils.FilterEventsOU(LogEvents, "GroupsSharingSettingsProto allow_unlisted_groups", utils.TopLevelOU)
+    DefaultSafe := true
+    Events := utils.FilterEvents(LogEvents, "GroupsSharingSettingsProto allow_unlisted_groups", utils.TopLevelOU)
```
Comment (adhilto) [inline rego/Groups.rego:361]: Same as above, why the `FilterEvents` function?

File: rego/Groups.rego (hunk 1 above)
Comment (snarve) [reply rego/Groups.rego]: must have been a typo, reverted it

File: rego/Groups.rego (hunk 2 above)
Comment (snarve) [reply rego/Groups.rego]: reverted

File: rego/Meet.rego [hunk 3]
```diff
@@ -276,12 +276,16 @@ tests contains {
     "PolicyId": "GWS.MEET.4.1v0.2",
     "Criticality": "Shall",
     "ReportDetails": utils.NoSuchEventDetails(DefaultSafe, utils.TopLevelOU),
-    "ActualValue": "No relevant event in the current logs",
+    "ActualValue": concat("", [
+        "No relevant event in the current logs for the top-level OU, Test Top-Level OU. ",
+        "While we are unable to determine the state from the logs, the default setting ",
+        "is non-compliant; manual check recommended."
+    ]),
```
Comment (snarve) [inline rego/Meet.rego]: also reverted this back, since this was updated while testing an error

**Review by adhilto — APPROVED** [review]

File: Testing/RegoTests/gmail/gmail14_test.rego [hunk 4]
```diff
@@ -99,12 +99,12 @@ test_EmailAllowlist_Incorrect_V1 if {
 
     RuleOutput := [Result | some Result in Output; Result.PolicyId == PolicyId]
     count(RuleOutput) == 1
-    not RuleOutput[0].RequirementMet
+    RuleOutput[0].RequirementMet
     RuleOutput[0].NoSuchEvent
-    RuleOutput[0].ReportDetails == concat("", [
+    RuleOutput[0].ReportDetails ==  concat("", [
```
Comment (mitchelbaker-cisa) [inline Testing/RegoTests/gmail/gmail14_test.rego:104]:
```suggestion
    RuleOutput[0].ReportDetails == concat("", [
```

**Review by mitchelbaker-cisa — APPROVED** [review]

File: Testing/RegoTests/gmail/gmail14_test.rego (hunk 4 above)
Comment (snarve) [reply Testing/RegoTests/gmail/gmail14_test.rego]: done

**Pull request merged**